control-plane/pkg/apis/messaging/v1beta1/kafka_channel_validation.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/eventing/pkg/apis/eventing"
+	"knative.dev/pkg/apis"
+)
+
+func (c *KafkaChannel) Validate(ctx context.Context) *apis.FieldError {
+	errs := c.Spec.Validate(ctx).ViaField("spec")
+
+	// Validate annotations
+	if c.Annotations != nil {
+		if scope, ok := c.Annotations[eventing.ScopeAnnotationKey]; ok {
+			if scope != "namespace" && scope != "cluster" {
+				iv := apis.ErrInvalidValue(scope, "")
+				iv.Details = "expected either 'cluster' or 'namespace'"
+				errs = errs.Also(iv.ViaFieldKey("annotations", eventing.ScopeAnnotationKey).ViaField("metadata"))
+			}
+		}
+	}
+
+	return errs
+}
+
+func (cs *KafkaChannelSpec) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
```

Comment:
If you change the return to a [named return value](https://yourbasic.org/golang/named-return-values-parameters/) you won't need that line!

Reply:
:-)  Looks like that's some good PRs to eventing _and_ the upstream version of the channel API

Reply:
I honestly don't like the named return values. Keep it this way and improve later if you like?